docs/tools/platformio.mdx

```diff
@@ -0,0 +1,124 @@
+
+### Definition for Luos engine Library
+
+| Definition  |                                            Description |                                                                           Value                                                                            |
+| :---------: | -----------------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   LUOSHAL   | Defines the HAL family to use with Luos engine library | <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow"> Luos HAL <IconExternalLink width="10" /></a> |
+| NOTELEMETRY |                                     Disables telemetry |                                                                       On by default                                                                        |
+
+Every Luos project needs to have a _node_config.h_ file allowing to configure Luos engine's hardware dependencies and configure the library. See the [basics concept page](/docs/luos-technology/basics/orga) for more information.
+
+Here is a _platformio.ini_ project example:
+
+```AsciiDoc
+[env:nucleo_f072rb]
+platform = ststm32
+board = nucleo_f072rb
+framework = stm32cube
+
+build_unflags = -Os
+build_flags =
+    -include node_config.h
+    -DLUOSHAL=STM32F0
+    -O1
+lib_deps =
+    luos_engine
+```
+
+This project in the examples is Luos engine dependent. PlatformIO will download the last version of the library Luos engine from its registry. The file _node_config.h_ allowing users to configure Luos parameters and the hardware definition to be added to the project.
+
+<div align="center">
+  <Image src="/img/telemetry.png" darkSrc="/img/telemetry.png" />
+</div>
+
+You must define the <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow">HAL family <IconExternalLink width="10" /></a> corresponding to your MCU. See the [compatibility page](/docs/compatibility/mcu_demoboard) to find the HAL family you must use for your MCU. During the build, a message appearing in red means a problem has occurred. It happens if you forget a build flag in your the file _platformio.ini_.
+
+<div align="center">
+  <Image src="/img/telemetry-1.png" darkSrc="/img/telemetry-1.png" />
+</div>
+
+By default, the telemetry is enabled and send anonymous information to Luos (only for improving Luos engine experience). To disable it, place the flag `NOTELEMETRY` in _platformio.ini_ the same way that the `LUOSHAL` flag.
+
+```AsciiDoc
+...
+
+build_flags =
+    -include node_config.h
+    -DLUOSHAL=STM32F0
+    -DNOTELEMETRY
+    -O1
+
+...
+```
+
+### Definition for the tools gate and pipe
```

Comment:
This part is better than the first one. More simple.
You should keep this format for the first one or make a first section about the general concept then make tables for every lib having a script.

Reply:
simpler for me because may concept and the way to understand this part is described above